client/src/App.test.js

```diff
@@ -1,14 +1,55 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { mount, shallow } from 'enzyme';
+import { sign } from 'jsonwebtoken';
 import App from './App';
 
 describe('<App/>', () => {
   test('renders without crashing', () => {
     shallow(<App />);
   });
 
-  test('should contain loggedInUser state', () => {
+  test('should contain user state', () => {
     const wrapper = shallow(<App />);
-    expect(wrapper.state().loggedInUser).toBeDefined();
+    expect(wrapper.state().user).toBeDefined();
+  });
+
+  test('should login user if there is a valid token and is not expired', () => {
+    const mockToken = sign({ username: 'abc', id: '123' }, 'secret', { expiresIn: 60 * 60 });
+    localStorage.setItem('token', mockToken);
+    const wrapper = mount(<App />);
+    expect(wrapper.state().user).toEqual({ username: 'abc', id: '123' });
+  });
+
+  test('should remove token from localstorage if token is invalid', () => {
+    localStorage.setItem('token', 'invalidtoken');
+    const wrapper = mount(<App />);
+    expect(wrapper.state().user).toBe(null);
```

Comment:
wrap this to checkUnAuthenticatedUser() function and use it all the test cases
```suggestion
    const checkUnAuthenticatedUser = ({ state }) => {
       expect(wrapper.state().user).toBe(null);
       expect(localStorage.removeItem).toHaveBeenCalledWith('token');
    }
    
```